termuic/termuiConsoleGrid.go

```diff
@@ -0,0 +1,170 @@
+
+	tcg.tSyncInfo = widgets.NewTable()
+	tcg.tSyncInfo.Rows = [][]string{
+		{"", "", "", ""},
+	}
+
+	tcg.lInstanceInfo = widgets.NewList()
+	tcg.lChainInfo = widgets.NewList()
+}
+
+func (tcg *termuiConsoleGrid) setupGrid() {
+	tcg.grid = ui.NewGrid()
+
+	tcg.grid.Set(
+		ui.NewRow(completRow/7, tcg.pPublicKey),
+		ui.NewRow(completRow/10, tcg.tSyncInfo),
+		ui.NewRow(completRow/3,
+			ui.NewCol(completCol/2,
+				ui.NewRow(completRow/4, tcg.pShardId),
+				ui.NewRow(completRow/4, tcg.pCountConsensus),
+				ui.NewRow(completRow/4, tcg.pCountLeader),
+				ui.NewRow(completRow/4, tcg.pCountAcceptedBlocks),
+			),
+			ui.NewCol(completCol/2,
+				ui.NewRow(completRow/2, tcg.pTxPoolLoad),
+				ui.NewRow(completRow/2, tcg.pNumConnectedPeers),
+			),
+		),
+		ui.NewRow(completRow/3, tcg.lLog),
+	)
+}
+
+func (tcg *termuiConsoleGrid) SetupBigLogGrid() {
+	tcg.grid = ui.NewGrid()
+
+	tcg.grid.Set(
+		ui.NewRow(1.0, tcg.lLog),
+	)
+}
+
+//Grid returns all console widgets that needs to be displayed
+func (tcg *termuiConsoleGrid) Grid() *ui.Grid {
+	return tcg.grid
+}
+
+func (tcg *termuiConsoleGrid) PrepareListWithLogsForDisplay(logData []string) {
+	tcg.lLog.Title = "Log info"
+	tcg.lLog.Rows = logData
+	tcg.lLog.TextStyle = ui.NewStyle(ui.ColorYellow)
+	tcg.lLog.WrapText = false
+
+	return
+}
+
+func (tcg *termuiConsoleGrid) PreparePublicKeyForDisplay(publicKey string) {
+	tcg.pPublicKey.Title = fmt.Sprintf("PublicKey")
+	tcg.pPublicKey.Text = fmt.Sprintf(publicKey)
+
+	return
+}
+
+func (tcg *termuiConsoleGrid) PrepareShardIdForDisplay(shardId int) {
+	//Check if node is in meta chain
+	//Node is in meta chain if shardId is equals with max uint32 value
+	if shardId == int(^uint32(0)) {
+		tcg.pShardId.Text = "ShardId: meta"
+	} else {
+		tcg.pShardId.Text = fmt.Sprintf("ShardId: %v", shardId)
+	}
+
+	return
+}
+
+func (tcg *termuiConsoleGrid) PrepareTxPoolLoadForDisplay(txPoolLoad int) {
+	tcg.pTxPoolLoad.Text = fmt.Sprintf("Tx pool load: %v", txPoolLoad)
+
+	return
+}
+
+func (tcg *termuiConsoleGrid) PrepareNumConnectedPeersForDisplay(numConnectedPeers int) {
+	tcg.pNumConnectedPeers.Text = fmt.Sprintf("Num connected peers: %v", numConnectedPeers)
+
+	return
+}
+
+func (tcg *termuiConsoleGrid) PrepareConsensusInformationsForDisplay(countConsensus int, countLeader int, acceptedBlocks int) {
+	tcg.pCountConsensus.Text = fmt.Sprintf("Count consensus group: %v", countConsensus)
+	tcg.pCountLeader.Text = fmt.Sprintf("Count leader: %v", countLeader)
+	tcg.pCountAcceptedBlocks.Text = fmt.Sprintf("Accepted blocks: %v", acceptedBlocks)
+
+	return
+}
+
+func (tcg *termuiConsoleGrid) PrepareSyncInfoForDisplay(nonce, currentRound, synchronizedRound, isSyncing int) {
+	isSyncingS := "Synchronized"
+	tcg.tSyncInfo.TextStyle = ui.NewStyle(ui.ColorWhite)
+	tcg.tSyncInfo.RowSeparator = true
+	tcg.tSyncInfo.BorderStyle = ui.NewStyle(ui.ColorWhite)
+	tcg.tSyncInfo.RowStyles[0] = ui.NewStyle(ui.ColorWhite, ui.ColorGreen, ui.ModifierBold)
+
+	if isSyncing == 1 {
```

Comment:
IsSynching is integer? Why?

Reply:
The name of variable was changed in syncStatus because lasted name was confusing. Type of variable is int.